lib/samson/form_builder.rb

```diff
@@ -50,8 +52,12 @@ def actions(delete: false, &block)
 
     # html regexp input ... allows blank values, does not know case insensitive so use a-zA-Z, does not know \A\z
     # FYI: could also read from validations: _validators[attribute].first.options[:with]
+    # js patterns need to match the full inout "/foo" will not match "/foobar" ... so we enforce \A + \z
     def translate_regex_to_js(pattern)
-      pattern.source.sub('\\A', '^').sub('\\z', '$') if pattern
+      pattern = pattern.source
+      raise ArgumentError, "pattern needs \\A" unless pattern.sub!('\\A', '^')
```

Comment:
looks like rails knows how to do this :D https://github.com/rails/rails/blob/b67043393b5ed6079989513299fe303ec3bc133b/actionpack/lib/action_dispatch/routing/inspector.rb#L42